Intersect.Client/Interface/Game/DescriptionWindows/ItemDescriptionWindow.cs

```diff
@@ -0,0 +1,478 @@
+
+                case ItemTypes.Consumable:
+                    SetupConsumableInfo();
+                    break;
+
+                case ItemTypes.Spell:
+                    SetupSpellInfo();
+                    break;
+
+                case ItemTypes.Bag:
+                    SetupBagInfo();
+                    break;
+            }
+
+            // if we've been passed a value label, set that up.
+            if (!string.IsNullOrWhiteSpace(mValueLabel))
+            {
+                SetupValueInfo();
+            }
+
+            // Resize the container, correct the display and position our window.
+            CorrectDisplay();
+            PositionWindow();
+
+            // If a spell, also display the spell description!
+            if (mItem.ItemType == ItemTypes.Spell)
+            {
+                mSpellDescWindow = new SpellDescriptionWindow(mItem.SpellId, mX, mContainer.Bottom);
+            }
+        }
+
+        protected void SetupHeader()
+        {
+            // Create our header, but do not load our layout yet since we're adding components manually.
+            mHeader = AddHeader();
+
+            // Set up the icon, if we can load it.
+            var tex = Globals.ContentManager.GetTexture(Framework.Content.TextureType.Item, mItem.Icon);
+            if (tex != null)
+            {
+                mHeader.SetIcon(tex, mItem.Color);
+            }
+
+            // Set up the header as the item name.
+            CustomColors.Items.Rarities.TryGetValue(mItem.Rarity, out var rarityColor);
+            var name = !string.IsNullOrWhiteSpace(mTitleOverride) ? mTitleOverride : mItem.Name;
+            mHeader.SetTitle(name, rarityColor ?? Color.White);
+
+            // Set up the description telling us what type of item this is.
+            // if equipment, also list what kind.
+            Strings.ItemDescription.ItemTypes.TryGetValue((int) mItem.ItemType, out var typeDesc);
+            if (mItem.ItemType == ItemTypes.Equipment)
+            {
+                var equipSlot = Options.Equipment.Slots[mItem.EquipmentSlot];
+                var extraInfo = equipSlot;
+                if (mItem.EquipmentSlot == Options.WeaponIndex && mItem.TwoHanded)
+                {
+                    extraInfo = $"{Strings.ItemDescription.TwoHand} {equipSlot}";
+                }
+                mHeader.SetSubtitle($"{typeDesc} - {extraInfo}", Color.White);
+            }
+            else
+            {
+                mHeader.SetSubtitle(typeDesc, Color.White);
+            }
+
+            // Set up the item rarity label.
+            Strings.ItemDescription.Rarity.TryGetValue(mItem.Rarity, out var rarityDesc);
+            mHeader.SetDescription(rarityDesc, rarityColor ?? Color.White);
+
+            mHeader.SizeToChildren(true, false);
+        }
+
+        protected void SetupItemLimits()
+        {
+            // Gather up what limitations apply to this item.
+            var limits = new List<string>();
```

Comment:
Its a objective and simple code, but maybe there is another ways to do this.
I guess you can use an Dictionary<bool, string> instead, add every bool with their respective StringResource and then run a ForEach adding to limits.
I another way, adding a little more complexity, you can create an Annotation just like "DescriptionData" and add it to this boolean properties in item class. So you can add data like "DescriptionType = ItemType.Limit" to identify limit properties like those, "DescriptionResource = String.ItemDescription.Banked" for description string, everything you need to make this and another process fully automatic. And if you dont want to add client~side data only to the ItemBase since its on Core project, you can create a new class just for that.

Its not a bad code actually. Just some suggestions in case of you consider any of them :P